docs/integrations/sources/postgres/postgres-troubleshooting.md

```diff
@@ -146,14 +151,123 @@ The connector waits for the default initial wait time of 5 minutes (300 seconds)
 
 If you know there are database changes to be synced, but the connector cannot read those changes, the root cause may be insufficient waiting time. In that case, you can increase the waiting time (example: set to 600 seconds) to test if it is indeed the root cause. On the other hand, if you know there are no database changes, you can decrease the wait time to speed up the zero record syncs.
 
-### (Advanced) WAL disk consumption and heartbeat action query
+### (Advanced) Resolving sync failures due to WAL disk consumption
+
+<!-- Applicable to other methods? -->
+
+When using the `Read Changes using Write-Ahead Log (CDC)` update method, you might encounter a situation where your initial sync is successful, but further syncs fail. You may also notice that the `confirmed_flush_lsn` column of the server's `pg_replication_slots` view doesn't advance as expected.
+
+This is a general issue that affects databases, schemas, and tables with small transaction volumes. There are complexities in the way PostgreSQL disk space can be consumed by WAL files, and these can cause issues for the connector. Airbyte's connector depends on Debezium. These complexities are outlined in [their documentation](https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-wal-disk-space), if you want to learn more.
+
+To fix the issue for both primary and replica:
```

Comment:
Currently we use this is a fix as a workaround to help the error when reading against replica. 

pg connector already provides a config that allows to specify a heartbeat query. For primary or standaline, I would still recommend using the heartbeat query and heartbeat table approach. I can provide more details on that approach. They are the same as this workaround, but they sounds more native to Debezium :-)

Reply:
I think the latest set of changes will address this comment.